compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/todo-global-load-cached.tsx

```diff
@@ -0,0 +1,17 @@
+import {Stringify} from 'shared-runtime';
+import {makeArray} from 'shared-runtime';
+
+/**
+ * Here, we don't need to memoize Stringify as it is a read off of a global.
+ * TODO: in PropagateScopeDeps (hir), we should produce a sidemap of global rvals
+ * and avoid adding them to `temporariesUsedOutsideDefiningScope`.
+ */
```

Comment:
Hmm, I don't think we're "memoizing" Stringify here. Instead, it's inside the mutable range of `arr` and therefore the temporary is actually used outside its declaring scope. 

This would be fixed by InstructionReordering, since it would move the LoadGlobal later to just before the JSX element, getting it out of the scope.